server/base/src/main/java/org/apache/accumulo/server/ServerInfo.java

```diff
@@ -0,0 +1,285 @@
+  }
+
+  @Override
+  public AuthenticationToken getAuthenticationToken() {
+    return getCredentials().getToken();
+  }
+
+  @Override
+  public boolean saslEnabled() {
+    return getServerConfFactory().getSystemConfiguration()
+        .getBoolean(Property.INSTANCE_RPC_SASL_ENABLED);
+  }
+
+  @Override
+  public Properties getProperties() {
+    Properties properties = new Properties();
+    properties.setProperty(ClientProperty.INSTANCE_ZOOKEEPERS.getKey(), getZooKeepers());
+    properties.setProperty(ClientProperty.INSTANCE_ZOOKEEPERS_TIMEOUT.getKey(),
+        Integer.toString(getZooKeepersSessionTimeOut()));
+    properties.setProperty(ClientProperty.INSTANCE_NAME.getKey(), getInstanceName());
+    ClientProperty.setAuthenticationToken(properties, getAuthenticationToken());
+    properties.setProperty(ClientProperty.AUTH_PRINCIPAL.getKey(), getPrincipal());
+    return properties;
+  }
+
+  public String getInstanceName() {
+    return instanceName;
+  }
+
+  public Credentials getCredentials() {
+    return SystemCredentials.get(getInstanceID());
+  }
+
+  public Connector getConnector(String principal, AuthenticationToken token)
+      throws AccumuloSecurityException, AccumuloException {
+    return Connector.builder().forInstance(instanceName, zooKeepers).usingToken(principal, token)
+        .withZkTimeout(zooKeepersSessionTimeOut).build();
+  }
+
+  public synchronized ServerConfigurationFactory getServerConfFactory() {
+    if (serverConfFactory == null) {
+      serverConfFactory = new ServerConfigurationFactory(this);
+    }
+    return serverConfFactory;
+  }
+
+  public String getApplicationName() {
+    Objects.requireNonNull(applicationName);
+    return applicationName;
+  }
+
+  public String getApplicationClassName() {
+    Objects.requireNonNull(applicationClassName);
+    return applicationName;
+  }
+
+  public String getHostname() {
+    Objects.requireNonNull(hostname);
+    return hostname;
+  }
+
+  public void teardownServer() {
+    DistributedTrace.disable();
+  }
+
+  public List<String> getMasterLocations() {
+
+    String masterLocPath = ZooUtil.getRoot(instanceID) + Constants.ZMASTER_LOCK;
+
+    OpTimer timer = null;
+
+    if (log.isTraceEnabled()) {
+      log.trace("tid={} Looking up master location in zoocache.", Thread.currentThread().getId());
+      timer = new OpTimer().start();
+    }
+
+    byte[] loc = ZooLock.getLockData(zooCache, masterLocPath, null);
+
+    if (timer != null) {
+      timer.stop();
+      log.trace("tid={} Found master at {} in {}", Thread.currentThread().getId(),
+          (loc == null ? "null" : new String(loc, UTF_8)),
+          String.format("%.3f secs", timer.scale(TimeUnit.SECONDS)));
+    }
+
+    if (loc == null) {
+      return Collections.emptyList();
+    }
+
+    return Collections.singletonList(new String(loc, UTF_8));
+  }
+
+  public String getRootTabletLocation() {
+    String zRootLocPath = ZooUtil.getRoot(instanceID) + RootTable.ZROOT_TABLET_LOCATION;
```

Comment:
Is this dupe of client context code?

Reply:
Yes. Fixed now. Only exists in ClientContext.